trade-finance-manager-api/src/v1/services/deal-cancellation/deal-cancellation.service.test.ts

```diff
@@ -0,0 +1,154 @@
+          formattedFacilitiesList: mockFormattedFacilities,
+          ukefDealId,
+        });
+      });
+    });
+
+    describe('when effective date is today', () => {
+      const aDealCancellation = (): TfmDealCancellation => ({ reason: 'a reason', bankRequestDate: today.valueOf(), effectiveFrom: today.valueOf() });
+
+      it('calls sendTfmEmail with the correct parameters', async () => {
+        // Arrange
+        const dealCancellation = aDealCancellation();
+
+        // Act
+        await DealCancellationService.submitDealCancellation(dealId, dealCancellation);
+        // Assert
+        expect(sendTfmEmail).toHaveBeenCalledTimes(1);
+        expect(sendTfmEmail).toHaveBeenCalledWith(CANCEL_DEAL_PAST_DATE, mockPimEmailAddress, {
+          cancelReason: dealCancellation.reason,
+          bankRequestDate: format(today, 'dd MMMM yyyy'),
+          effectiveFromDate: format(today, 'dd MMMM yyyy'),
+          formattedFacilitiesList: mockFormattedFacilities,
+          ukefDealId,
+        });
+      });
+    });
+  });
+
+  describe('sendDealCancellationEmail', () => {
+    describe('when effective date is in the future', () => {
+      const aDealCancellation = (): TfmDealCancellation => ({
+        reason: 'a reason',
+        bankRequestDate: today.valueOf(),
+        effectiveFrom: tomorrow.valueOf(),
+      });
+
+      it('calls sendTfmEmail with the correct parameters', async () => {
+        // Arrange
+        const dealCancellation = aDealCancellation();
+
+        // Act
+        await DealCancellationService.sendDealCancellationEmail(ukefDealId, aDealCancellation(), ukefFacilityIds);
+
+        // Assert
+        expect(sendTfmEmail).toHaveBeenCalledTimes(1);
+        expect(sendTfmEmail).toHaveBeenCalledWith(CANCEL_DEAL_FUTURE_DATE, mockPimEmailAddress, {
+          cancelReason: dealCancellation.reason,
+          bankRequestDate: format(today, 'dd MMMM yyyy'),
+          effectiveFromDate: format(tomorrow, 'dd MMMM yyyy'),
+          formattedFacilitiesList: mockFormattedFacilities,
+          ukefDealId,
+        });
+      });
+    });
+
+    describe('when effective date is today', () => {
+      const aDealCancellation = (): TfmDealCancellation => ({ reason: 'a reason', bankRequestDate: today.valueOf(), effectiveFrom: today.valueOf() });
+
+      it('calls sendTfmEmail with the correct parameters', async () => {
+        // Arrange
+        const dealCancellation = aDealCancellation();
+
+        // Act
+        await DealCancellationService.sendDealCancellationEmail(ukefDealId, aDealCancellation(), ukefFacilityIds);
+
+        // Assert
+        expect(sendTfmEmail).toHaveBeenCalledTimes(1);
+        expect(sendTfmEmail).toHaveBeenCalledWith(CANCEL_DEAL_PAST_DATE, mockPimEmailAddress, {
+          cancelReason: dealCancellation.reason,
+          bankRequestDate: format(today, 'dd MMMM yyyy'),
+          effectiveFromDate: format(today, 'dd MMMM yyyy'),
```

Comment:
Would be nice to use constants for `'dd MMMM yyyy'` - not a blocker, we have 42 instances of this, should be done in a different PR